### ISSUE TITLE
Pattern matching throws false positive F821 when the value comes from the pattern

### ISSUE THREAD

**Issue by Vegemash** [charliermarsh/ruff#3103]
Woo pattern matching! Thank you for supporting 99% of our codebase, here's some bugs :P

```python
class TheEyesHaveIt:
    def __init__(self, i=None, eye=None):
        self.i = i
        self.eye = eye


def match_the_thing(thing: TheEyesHaveIt) -> None:
    match thing:
        case TheEyesHaveIt(i=value) if value:
            print(f"The i's have it at {value}")
        case TheEyesHaveIt(eye=value) if value:
            print(f"The eyes have it at {value}")


a = TheEyesHaveIt(eye="foo")
b = TheEyesHaveIt(i="bar")
match_the_thing(a)
match_the_thing(b)
```

All those `value`s are marked as undefined names "F821 Undefined name \`value\`".

**Comment by charliermarsh** [comment]
I think these were fixed by #3098, but I'll double check.

I'll probably cut a new release today to get that out.
